app/scripts/services/Citations.svc.js

```diff
@@ -1,14 +1,52 @@
 'use strict';
 
-angular.module('yourStlCourts').factory('Citations', function ($resource) {
+angular.module('yourStlCourts').factory('Citations', function ($resource,$q) {
   var CitationResource = $resource('citations/:id');
 
   function find(parameters) {
-    return CitationResource.get(parameters).$promise;
+    var deferred = $q.defer();
+    CitationResource.get(parameters,function(citations){
```

Comment:
I wasn't suggesting you remove the date conversion at all....you still don't need the `defer`. Like this...

```
CitationResource.get(parameters,function(citations){
  for(var count = 0; count < citations.length; count++){
    convertCitationDateStringsToDates(citations[count]);
  }

  return citations;
});
```

No `$q.defer` at all.